### PR TITLE
testcases and fixes for vmware_guest_facts

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/vmware/vmware_guest_facts.py
test/integration/targets/vmware_guest_facts/aliases
test/integration/targets/vmware_guest_facts/tasks/main.yml

##### ANSIBLE VERSION
```
2.4devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
ready_for_review

File: lib/ansible/modules/cloud/vmware/vmware_guest_facts.py [hunk 1]
```diff
@@ -121,10 +131,7 @@ def getvm(self, name=None, uuid=None, folder=None):
         if uuid:
             vm = find_vm_by_id(self.content, vm_id=uuid, vm_id_type="uuid")
         elif folder:
-            # Build the absolute folder path to pass into the search method
-            if not self.params['folder'].startswith('/'):
-                self.module.fail_json(msg="Folder %(folder)s needs to be an absolute path, starting with '/'." % self.params)
-            searchpath = '%(datacenter)s%(folder)s' % self.params
+            searchpath = self.params['folder']
```
Comment (alikins) [inline lib/ansible/modules/cloud/vmware/vmware_guest_facts.py:134]: Could this cause backwards compatibility issues?  If someone was relying on the previous behavior of adding the datacenter prefix to searchpath?

I think the approach in this pr is better and more explicit. But I'm not sure if the change will introduce api differences from the 2.3 version though. Would users see a behavior change?

**Review by alikins — CHANGES_REQUESTED** [review]
Change LGTM to me, especially if there are no backwards compat problems. 

So mostly just requesting confirmation about the compatibility.

File: lib/ansible/modules/cloud/vmware/vmware_guest_facts.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_guest_facts.py:134]: I don't think there will be any backwards compatibility issues with this change. User would not see any behavior change as we are internally changing searchpath. I have added examples for folder path in documentation section which will give a clear idea about this change. 
@jctanner Would you like add something to this ?

**Comment by jctanner** [comment]
This is replicating the same logic from vmware_guest, so we need to merge it.

**Comment by Akasurde** [comment]
@jctanner @alikins Thanks for reviews and merge.

**Pull request merged**